ApiData.ts

```diff
@@ -0,0 +1,266 @@
+/** Debounce utility function to limit function execution frequency */
+function debounce<F extends (...args: any) => any>(func: F, waitFor: number) {
+	let timeout: number;
+
+	return (...args: Parameters<F>): Promise<ReturnType<F>> => {
+		clearTimeout(timeout);
+
+		return new Promise((resolve) => {
+			timeout = setTimeout(() => {
+				resolve(func(...args));
+			}, waitFor);
+		});
+	};
+}
+
+/** Constants for grid calculation
+* GRID_RATIO represents the ratio of the grid's height to the window's height.
+*/
+const GRID_RATIO = 9 / 20;
+const ROW_HEIGHT = 16;
+
+/** manage data and settings on the grid */
+class ApiData {
+
+	pageSize: number;
+	currentPage: number = 1;
+	data: GridData[] = [];
+	totalItems: number = 0;
+	columnNames: ColumnName[] = [];
+	maxGridHeight: number = 0;
+	firstVal: number = 0;
+	lastVal: number = -1;
+	maxRange: number;
+
+	constructor(pageSize: number) {
+		this.pageSize = pageSize;
+		this.maxRange = 0;
+
+	}
+
+	/** Initialize method to set up the grid */
+	initialize(): Promise<void> {
+		this.adjustGridHeight();
+		return this.recordCount()
+			.then(() => this.fetchColumns())
+			.then(() => this.fetchRecords())
+			.then(() => this.setupControls());
+	}
+	/** Fetch total record count from the server */
+	recordCount(): Promise<void> {
+		return this.fetchData('http://localhost:2050/recordCount')
+			.then((response: any) => {
+				const totalItems = response;
+				this.totalItems = totalItems;
+			})
+			.catch(error => {
+				console.error('Failed to fetch record count:', error);
+				alert('Failed to fetch record count.');
+			});
+	}
+
+	/** fectch column names */
+	fetchColumns(): Promise<void> {
+		return this.fetchData('http://localhost:2050/columns')
+			.then((response: any) => {
+				const res = JSON.parse(response);
+				this.columnNames = res.map((columnName: any) => ({ name: columnName }));
+				this.data = new Array<GridData>(this.columnNames.length);
+			})
+			.catch(() => {
+				throw ('Failed to fetch columns.');
+			});
+	}
+
+	/** get records from API for fetch and search functionality */
+	fetchAndProcessRecords(from: number, to: number): Promise<GridData[]> {
+		$('#spinner').show();
+		$('#grid').hide();
+
+		return this.fetchData(`http://localhost:2050/records?from=${from}&to=${to}`)
+			.then((response: any) => {
+				const res = JSON.parse(response);
+				const processedData = res.map((record: any) => {
+					const obj: GridData = {};
+					for (let j = 0; j < this.columnNames.length && j < record.length; j++) {
+						obj[this.columnNames[j].name] = record[j];
+					}
+					return obj;
+				});
+				$('#spinner').hide();
+				$('#grid').show();
+				return processedData;
+			})
+			.catch(() => {
+				throw ('Failed to fetch records');
+			});
+	}
+
+	/** fetch records from api */
+	fetchRecords(): Promise<void> {
```

Comment:
This function name and comment does not really match up what the function is doing... Yes, it is fetching records, but it is doing that by calling `fetchAndProcessRecords` and then it is also displaying the records and updating the page info. Just feel like if someone read the function name and  comment, they would assume it is doing only that, and not also something else.

Please just add a bit more to the function name and comment.